src/caret_analyze/architecture/architecture_loaded.py

```diff
@@ -642,11 +667,15 @@ def __init__(
     def get_node_path(
         node_paths: Sequence[NodePathStruct],
         publisher_topic_name: str,
-        subscription_topic_name: str
+        subscription_topic_name: str,
+        subscription_construction_order: int,
+        publisher_construction_order: int
```

Comment:
Is there any reason for pub->sub->sub->pub order of this arguments? I think pub->sub->pub->sub is better to avoid confusion.